tools/syz-declextract/run.go

```diff
@@ -0,0 +1,62 @@
+// Copyright 2017 syzkaller project authors. All rights reserved.
+// Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+package main
+
+import (
+	"encoding/json"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/google/syzkaller/pkg/tool"
+)
+
+type compileCommand struct {
+	Arguments []string
+	Directory string
+	File      string
+	Output    string
+}
+
+func run(file string, out chan []byte) {
+	if !strings.HasSuffix(file, ".c") {
+		out <- nil
+		return
+	}
+
+	cmd := exec.Command("syz-declextract", "-p", os.Args[1], file)
+	stdout, err := cmd.Output()
+	if err != nil {
+		tool.Fail(err)
+	}
+	out <- stdout
+}
+
+func main() {
+	if len(os.Args) < 2 {
+		tool.Failf("Please input path to compile_commands.json")
+	}
+	var fileData []byte
+	var err error
+
+	if fileData, err = os.ReadFile(os.Args[1]); err != nil {
+		tool.Fail(err)
+	}
+
+	var c []compileCommand
+
+	if err = json.Unmarshal(fileData, &c); err != nil {
+		tool.Fail(err)
+	}
+
+	out := make(chan []byte, len(c))
+
+	for _, v := range c {
```

Comment:
This will start goroutines and syz-declextract for all files at once. If there are 5000 files, it will start 5000 subprocesses. It may lead to OOM, or just be slower due to excessive oversubscriptions.

In such cases it's better to create a fixed set of worker goroutines that will process requests from an inputs channel, see:
https://gobyexample.com/worker-pools
or:
https://github.com/google/syzkaller/blob/master/sys/syz-extract/extract.go#L100-L102

Reply:
Done!